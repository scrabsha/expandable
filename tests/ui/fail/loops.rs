#![allow(unused)]

#[expandable::expr]
macro_rules! a {
    () => {
        loop loop {}
    };
}

#[expandable::expr]
macro_rules! b {
    () => {
        while < 42 {}
    };
}

#[expandable::expr]
macro_rules! c {
    () => {
        for ! in in {}
    };
}

fn main() {}
