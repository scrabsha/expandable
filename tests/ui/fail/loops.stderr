error: Potentially invalid expansion. Expected a `{`.
 --> tests/ui/fail/loops.rs:6:14
  |
6 |         loop loop {}
  |              ^^^^

error: Potentially invalid expansion. Expected an identifier, `Self`, `_`, `self`, `super`.
  --> tests/ui/fail/loops.rs:13:17
   |
13 |         while < 42 {}
   |                 ^^

error: Potentially invalid expansion. Expected a pattern, an identifier, `&`, `-`, `..=`, `..` or 10 others.
  --> tests/ui/fail/loops.rs:20:13
   |
20 |         for ! in in {}
   |             ^
